rclcpp/include/rclcpp/clock.hpp

```diff
@@ -121,6 +123,18 @@ class Clock
     JumpHandler::post_callback_t post_callback,
     const rcl_jump_threshold_t & threshold);
 
+  /**
+   * Get current time from the time source specified by Clock::get_clock_type() const noexcept.
+   *
+   * \return RCUTILS_RET_OK if time was successfully retrieved
+   * \param now. A time point value pointer that will contain the
+   * current time from the underlying clock
+   * \throws anything rclcpp::exceptions::throw_from_rcl_error can throw.
+   */
+  RCLCPP_PUBLIC
+  rcutils_ret_t
```

Comment:
This should not return an `rcutils_ret_t`, this is a C++ function, if it is not `RCUTILS_RET_OK`, then you should throw an exception. In fact the documentation says it will throw an exception.

In which case, I think this function should take no arguments and should return an `rcutils_time_point_value_t` or a C++ equivalent, if there is an appropriate one.

Reply:
Imo the cost of reimplementing the macros outweighs making this function be more C++-like and throw. I would suggest to instead remove the throw completely from this signature and just return an `rcutils_ret_t`.

If we want to make it more clear that this function is only supposed to be used for the logging macro we could mangle the name to something like: `get_now_as_time_point_for_logging`.

Reply:
I read this comment after implementing the new macro. What are your thoughts about it now?